IPNI_HTTP_PROVIDER.md

```diff
@@ -0,0 +1,310 @@
+opportunities to make the transport mechanism swappable.
+
+## Table of Contents
+
+- [IPNI HTTP Provider](#ipni-http-provider)
+- [Motivation](#motivation)
+- [Background](#background)
+    - [Common Data Types](#common-data-types)
+    - [Encoding](#encoding)
+    - [Versioning](#versioning)
+    - [Namespacing](#namespacing)
+        - [Implicit Namespace Assumption](#implicit-namespace-assumption)
+        - [Custom Paths in Multiaddr](#custom-paths-in-multiaddr)
+- [Specification](#specification)
+    - [GET `/ipni/v1/ad/head`](#get-ipniv1adhead)
+        - [Response](#response)
+        - [Example](#example)
+    - [GET `/ipni/v1/ad/{CID}`](#get-ipniv1adcid)
+        - [Response](#response-1)
+        - [Example](#example-1)
+- [Security](#security)
+- [Future Considerations](#future-considerations)
+- [Related Resources](#related-resources)
+- [Copyright](#copyright)
+
+## Motivation
+
+The motivation behind formalizing the HTTP API specification for IPNI providers stems from the need for a standardized
+and well-defined protocol for interacting with HTTP-based index providers. Despite the existence of the HTTP protocol
+for providers from early on, there has been no formal definition of the API specific to IPNI. By establishing a clear
+and consistent API specification, we aim to achieve several goals:
+
+1. **Explicit Definition** -- A formal definition of the HTTP API for IPNI providers fills a crucial gap in the IPNI
+   specification. It provides a comprehensive and explicit outline of the required endpoints, request/response formats,
+   and behavior expected from providers. Such definition can then be referenced as part of the capability required by a
+   provider that aims to offer retrieval over HTTP.
+
+2. **Grouping of Capabilities** -- An implicit grouping of API endpoints is essential because HTTP servers running IPNI
+   capabilities often provide multiple functionalities. By organizing the API endpoints under a unified specification,
+   it becomes easier to understand and navigate the various capabilities offered by an IPNI provider while avoiding
+   conflict with other APIs such as the IPFS Gateway or Boost.
+
+3. **Interoperability and Swappable Transport** -- A formalized API specification facilitates future opportunities for
+   interoperability between different IPNI providers and enables the potential for swappable transport mechanisms. By
+   adhering to a common API standard, providers can ensure compatibility and seamless integration with other components
+   of the IPNI ecosystem, while reducing engineering footprint.
+
+Overall, the formal definition of the HTTP API for IPNI providers enhances clarity, promotes consistency, and opens up
+opportunities for innovation and collaboration within the IPNI network.
+
+## Background
+
+### Common Data Types
+
+The IPNI HTTP Provider API specification makes use of the following common data types:
```

Comment:
Your call, but if you're expecting others outside PL to read this, could be good to link to sources for more information.

Maybe use reference style links?